src/components/structures/auth/SoftLogout.js

```diff
@@ -63,10 +85,120 @@ export default class SoftLogout extends React.Component {
         });
     };
 
-    onLogin = () => {
-        dis.dispatch({action: 'start_login'});
+    async _initLogin() {
+        const client = MatrixClientPeg.get();
+        const loginViews = (await client.loginFlows()).flows.map(f => FLOWS_TO_VIEWS[f.type]);
+
+        const chosenView = loginViews.filter(f => !!f)[0] || LOGIN_VIEW.UNSUPPORTED;
```

Comment:
we can assume here that only the first flow is relevant as any terms & conditions should already be accepted?

Reply:
This is the first flow of flows we have views for, so it should be safe. It discards anything custom/unknown in the `.filter()` due to the line above. 

This also means that terms of service get lost, and they don't technically get served at this point yet (the `.loginFlows()` call is without auth, so the server doesn't know if the user needs to sign terms of service).